packages/api/amplify_api/test/amplify_api_config_test.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:convert';
+
+import 'package:amplify_api/src/amplify_api_config.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'test_data/fake_amplify_configuration.dart';
+
+void main() {
+  late EndpointConfig endpointConfig;
+
+  group('GraphQL Config', () {
+    const endpointType = EndpointType.graphQL;
+    const endpoint =
+        'https=//abc123.appsync-api.us-east-1.amazonaws.com/graphql';
+    const region = 'us-east-1';
+    const authorizationType = APIAuthorizationType.apiKey;
+
+    setUpAll(() async {
+      const config = AWSApiConfig(
+          endpointType: endpointType,
+          endpoint: endpoint,
+          region: region,
+          authorizationType: authorizationType);
+
+      endpointConfig = const EndpointConfig('GraphQL', config);
+    });
+
+    test('should return valid URI wtih null params', () async {
+      final uri = endpointConfig.getUri(path: null, queryParameters: null);
+      final expected = Uri.parse('$endpoint/');
+
+      expect(uri, equals(expected));
+    });
+
+    test('should returns invalid URI wtih params', () async {
+      final uri = endpointConfig
+          .getUri(path: 'random/path', queryParameters: {'key': 'value'});
+      final expected = Uri.parse('$endpoint/');
+
+      expect(uri, isNot(expected));
+    });
+  });
+
+  group('REST Config', () {
+    const endpointType = EndpointType.rest;
+    const endpoint =
+        'https=//abc123.appsync-api.us-east-1.amazonaws.com/graphql';
+    const region = 'us-east-1';
+    const authorizationType = APIAuthorizationType.apiKey;
```

Comment:
nit: please do IAM w REST for this mock as that's the only thing the CLI will do for you at the moment and api key not possible w REST.